robosherlock/src/filter/src/PointCloudFilter.cpp

```diff
@@ -104,7 +134,18 @@ class PointCloudFilter : public DrawingAnnotator
     pass.setFilterFieldName("z");
     pass.filter(*cloud_filtered);
 
-    cas.set(VIEW_CLOUD, *cloud_filtered);
+    pcl::PointCloud<PointT>::Ptr cloud(new pcl::PointCloud<PointT>);
+    if(transform_cloud) {
+        outInfo("Transforming matrix back (this will deeply hurt the CPU...");
```

Comment:
Alternatively you could check for NaN-s in filtered cloud and set their counterparts in cloud to NaN